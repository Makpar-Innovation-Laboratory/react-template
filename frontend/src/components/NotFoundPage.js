import React from "react";
import { history } from "../store/configureStore";

const NotFoundPage = () => {
  return (
    <div id="404-page-container">
      <h4>
        404 Page Not Found
      </h4>
      <button
        id="home-button"
        className="usa-button"
        aria-label="Click to return to Home Screen"
        value="Go Back To Home Page"
        onClick={() => history.push("/")}
      >
        Go Back To Home Page
      </button>
    </div>
  );
};

export default NotFoundPage;
