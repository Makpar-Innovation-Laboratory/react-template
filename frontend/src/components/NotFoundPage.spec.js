import React from "react";
import { shallow } from "enzyme";
import NotFoundPage from "./NotFoundPage";
import { isolateComponent } from "isolate-react";
import { history } from "../store/configureStore";

describe("JSX Tests :: <NotFoundPage /> Rendering", () => {

  test("Sanity Check Snapshot Rendering", () => {

    const wrapper = shallow(<NotFoundPage />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test("Home Button Correctly Navigates To Adjudicator Home Page", () => {
    const props = ({ location: "/NotFoundPage"});
    const wrapper = isolateComponent(<NotFoundPage {...props} />);
    const homeBtn = wrapper.findOne("#home-button");

    homeBtn.props.onClick();

    expect(history.location.pathname).toEqual("/");
  });
});
