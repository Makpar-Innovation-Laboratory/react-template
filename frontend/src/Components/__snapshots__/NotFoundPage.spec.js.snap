// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSX Tests :: <NotFoundPage /> Rendering Sanity Check Snapshot Rendering 1`] = `
"<div id=\\"404-page-container\\">
  <h4>
    404 Page Not Found
  </h4>
  <button id=\\"home-button\\" className=\\"usa-button\\" aria-label=\\"Click to return to Home Screen\\" value=\\"Go Back To Home Page\\" onClick={[Function: onClick]}>
    Go Back To Home Page
  </button>
</div>"
`;
